engine-nextgen/src/main/kotlin/graphql/nadel/enginekt/transform/NadelDeepRenameTransform.kt

```diff
@@ -377,19 +398,21 @@ internal class NadelDeepRenameTransform : NadelTransform<NadelDeepRenameTransfor
         return state.alias + TypeNameMetaFieldDef.name
     }
 
-    private fun getMatchingDeepRename(
+    /**
+     * Note: this can be null if the type condition was not met
+     */
+    private fun getMatchingDeepRenameInstruction(
         parentMap: JsonMap,
         state: State,
-        executionPlan: NadelExecutionPlan
-    ): NadelDeepRenameFieldInstruction {
+        executionPlan: NadelExecutionPlan,
+    ): NadelDeepRenameFieldInstruction? {
```

Comment:
I realized that this _can_ be null. Say not all of the object types have the hydration, then we might not have an instruction, and in that case we shouldn't do anything e.g.

```graphql
type Query {
    pet: Pet
}

interface Pet {
    name: String
}

type Dog implements Pet {
    name: String @renamed(from: ["details", "name"])
}

type Cat implements Pet {
    name: String
}
```

If we receive a `Cat`, we might error out.